### ISSUE TITLE
MSSQL Decimal Type Parameter Issue

### ISSUE THREAD

**Issue by j-franco-applaudo** [strongloop/loopback-connector-mssql#183]
MSSQL Fields that have a decimal type get extra decimals after an update or create. For instance, if you want to insert in a field named `position` that is `decimal(18,11)` a value of `0.90` you will get a number like `0.900000002212` in your database.

After trying to determinate if the issue was the database or the connector. I found out that the logic to generate the params use `mssql.Real` when defining the parameters.

For example, this a section of the query in my database executed by the connector:

```(@param1 int, @param2 int, @param3 int, @param4 int, @param5 int, @param6 int, @param7 int, @param8 real, @param9 int)UPDATE [dbo].```

where the `param8` is set to be real when it should be `decimal(18,12)`.

My JSON declaration for the field is something like this:
```
"position": {
      "type": "Number",
      "required": true,
      "length": null,
      "precision": 18,
      "scale": 12,
      "mssql": {
        "columnName": "Position",
        "dataType": "decimal",
        "dataLength": null,
        "dataPrecision": 18,
        "dataScale": 12,
        "nullable": "NO"
      }
    }
```
And if you take a look at the code the paramters type `mssql.Real` is assigned based on the value not the JSON declaration.

https://github.com/strongloop/loopback-connector-mssql/blob/b5332c06a0dc465bfc8ae44a3af13590fa6292c2/lib/mssql.js#L141-L144

Just to be sure that the problem was that line I changed it to this:

``` request.input('param' + (i + 1), mssql.Decimal(18, 12), params[i]);```

Now the decimals are inserted exactly as is intended. However, this will not fix the issue since there is no way to get the JSON information to select the correct types for the parameters in that section of the connector.

**Comment by j-franco-applaudo** [comment]
Hi @raymondfeng, Do you know if there is a workaround?. So far we are trying to use this connector in a production project, but this issue is a big blocker.

**Comment by aroper23** [comment]
I encountered the same issue with data being lost writing to a Decimal(15, 2) column. My solution was to update the executeSQL code in mssql.js to use the type mssql.Float instead of mssql.Real since it stores more information.

Perhaps one of the project contributors could update this in the project code? For now, I added a boot script to my project to patch the definition of executeSQL as the application starts.

[patchMSSqlConnector.js.txt](https://github.com/strongloop/loopback-connector-mssql/files/2525974/patchMSSqlConnector.js.txt)
